packages/sources/tiingo-test/src/endpoint/ws/crypto.ts

```diff
@@ -0,0 +1,69 @@
+import { WebSocketTransport } from '@chainlink/external-adapter-framework/transports/websocket'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { customSettings } from '../../config'
+import { PriceEndpointParams } from '@chainlink/external-adapter-framework/adapter'
+
+interface Message {
+  service: string
+  messageType: string
+  data: [string, string, string, string, number]
+}
+
+const tickerIndex = 1
+const priceIndex = 4
+
+type EndpointTypes = {
+  Request: {
+    Params: PriceEndpointParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: Message
+  }
+}
+
+let apiKey = ''
+export const wsTransport = new WebSocketTransport<EndpointTypes>({
+  url: (context) => {
+    apiKey = context.adapterConfig.API_KEY
+    return `${context.adapterConfig.WS_API_ENDPOINT}/crypto-synth`
+  },
```

Comment:
Instead of setting global variables, should we instead create a class that extends WebSocketTransport which can hold the API key? @alejoberardino opinions?

Reply:
I think a subclass would be cleaner, yeah. Another option is to expose the config to the builders, but maybe that's too much of a hassle

Reply:
Done